### PR TITLE
chore: run label action on prs

### PR DESCRIPTION
**Pull request by viceice**
## Changes
 Run label action on PR's via `pull_request_target` so we use the workflow from main and the token can write to fork PR's.

## Context
- #18836
- #18838
- https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request_target

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: .github/workflows/label-actions.yml [hunk 1]
```diff
@@ -5,11 +5,14 @@ on:
     types: [labeled]
   discussion:
     types: [labeled]
+  pull_request_target:
```
[the remark this answers is not in the document]
Comment (viceice) [reply .github/workflows/label-actions.yml:8]: We don't do any checkout.
We only run on labeled type, which can only triggered by us.
We explicit trim workflow permissions.
The workflow is always loaded from main branch, so not possible to inject something via PR.

**Review by secustor — APPROVED** [review]
LGTM, tough I have never used types selectors.

**Comment by rarkins** [comment]
What specifically do we need this for?

**Comment by viceice** [comment]
to use automated comments on pull request labels

**Review by rarkins — APPROVED** [review]

**Pull request merged**